### PR TITLE
fix(TDI-47538): tAzureAdlsGen2Output component converts BigDecimal to String

### PR DESCRIPTION
**Pull request by jzhao-talend**
for: https://jira.talendforge.org/browse/TDI-47538

tAzureAdlsGen2Output write and read bigdecimal type in parquet type

### CODE REVIEW

File: common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public static final String AVRO_LOGICAL_TYPE_TIME_MILLIS = "time-millis";
+
+    public static final String AVRO_LOGICAL_TYPE_TIMESTAMP_MILLIS = "timestamp-millis";
+
+    public static final String AVRO_LOGICAL_TYPE_DECIMAL = "decimal";
+
+    public static final String ERROR_UNDEFINED_TYPE = "Undefined type %s.";
+
+    public static final String STUDIO_LENGTH = "talend.studio.length";
```
Comment (clesaec) [inline common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java:33]: Why named this constant with "Studio" ? As it is used for Avro & Parquet for Pipeline deigner & studio.
Should be better with
`public static final String DECIMAL_LENGTH = "talend.decimal.length";`
for example ?

**Review by clesaec — APPROVED** [review]
Suggest one small change about constant Names.

File: common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java (hunk 1 above)
Comment (jzhao-talend) [reply common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java:33]: https://github.com/Talend/component-runtime/pull/551/files#diff-b2dc0f214b21ca9bf02841caeb3774827c739f648ee10bd74386fac92137366eR21-R27
should be the same with component-runtime

Comment (jzhao-talend) [reply common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java:33]: so component-runtime-di also needs to adjust to a generic name. event it only for the studio.
during avro convert, we need to get the decimal properties from the schema generated by component-runtime-di

Comment (clesaec) [reply common-stream-io/stream-avro/src/main/java/org/talend/components/common/stream/Constants.java:33]: ok, i see

**Comment by clesaec** [comment]
Just to be sure, could you add a test 'record to Avro' with an array of BigDecimal element

**Comment by jzhao-talend** [comment]
> 

what about the talend type are you talking about?
List or Object?

**Comment by clesaec** [comment]
I mean a org.apache.avro.Schema.Type.ARRAY of org.apache.avro.Schema.Type.BYTES that have Constants.AVRO_LOGICAL_TYPE_DECIMAL logical types.

**Review by clesaec — APPROVED** [review]

**Review by clesaec — APPROVED** [review]

**Pull request merged**